autoload/flog.vim

```diff
@@ -1012,6 +1012,190 @@ endfunction
+endfunction
+
+" See https://vi.stackexchange.com/questions/29056/how-to-find-first-item-that-satisfies-predicate/29059#29059
+function! flog#find_predicate(haystack, predicate) abort
+  return get(filter(copy(a:haystack), "a:predicate(v:val)"), 0, v:null)
+endfunction
+
+fu! flog#find_commit(state, commit_hash) abort
+  return flog#find_predicate(a:state.commits, {item -> flog#starts_with(a:commit_hash, item.short_commit_hash)})
+endfunction
+
+fu! flog#jump_to_commit(commit_hash) abort
+  let l:state = flog#get_state()
+  let l:commit = flog#find_commit(l:state, a:commit_hash)
+  if type(l:commit) != v:t_dict
+    let l:state.reflog = v:true
+    call flog#populate_graph_buffer()
+    let l:commit = flog#find_commit(l:state, a:commit_hash)
+  endif
+  let l:index = index(l:state.commits, l:commit)
+  let l:index = min([max([l:index, 0]), len(l:state.commits) - 1])
+  let l:line = index(l:state.line_commits, l:state.commits[l:index]) + 1
+  if l:line >= 0
+    exec l:line
+    call flog#to_commit_start()
+  endif
+endfunction
+
+fu! flog#get_short_commit_hash() abort
+  let l:state = flog#get_state()
+  let l:current_commit = flog#get_commit_at_line()
+  if type(l:current_commit) != v:t_dict
+    return
+  endif
+  return l:current_commit.short_commit_hash
+endfunction
+
+fu! flog#get_full_commit_hash() abort
+  let l:short = flog#get_short_commit_hash()
+  if type(l:short) != v:t_string
+    return v:null
+  endif
+  return fugitive#RevParse(flog#get_short_commit_hash())
+endfunction
+
+" Returns one of the elements of the list.
+" If it has been called before with a commit
+" present in the list, or has previously returned
+" an element of the current list, *that* element will be
+" returned. In addition, if the current commit is
+" not in the list of prior inputs and outputs,
+" that list is cleared. Hence there can only be one
+" matching element.
+" Use this as a replacement for just doing commit_list[0],
+" giving the current commit as an additional input. 
+" In the base case, it just returns
+" the first element. But the most recent inputs and outputs 
+" of this function are given priority before any others,
+" giving the function 'stability' when navigating up/down
+" a directed acyclic graph
+fu! flog#stable_select(commit_list, current_commit) abort
+  if index(g:flog_visited_commits, a:current_commit) == -1
+    let g:flog_visited_commits = [a:current_commit]
+  endif
+  let pick = flog#find_predicate(a:commit_list, {target_commit -> index(g:flog_visited_commits, target_commit) != -1})
+  if pick == v:null
+    let pick = a:commit_list[0]
+  endif
+  call add(g:flog_visited_commits, l:pick)
+  return pick
+endfunction
+
+fu! flog#jump_up_N_parents(amount) abort
+  let l:current_commit = flog#get_full_commit_hash()
+  if type(l:current_commit) != v:t_string
+    return v:null
+  endif
+  let c = 0
+  while c < a:amount
+    let l:parent_commit = system("git rev-list --parents -n 1 " . l:current_commit)
```

Comment:
Git commands should use `flog#get_fugitive_git_command()`

Reply:
You might want to have a quick skim of the other utility functions while you're at it, I still have to check but I think there might be more functions you can use.